src/geo-map-direction-service-google.ts

```diff
@@ -40,6 +40,10 @@ export class GeoMapDirectionServiceGoogle
       directionsService.route(config, results => {
         const [route] = results.routes;
 
+        if (!route || !Array.isArray(route.legs)) {
```

Comment:
Do a separate check for `route.legs`:

```suggestion
  if (typeof route !== 'object' || route === null) {
     return reject(new Error(`No route object found`));
  }
  
  if (Array.isArray(route.legs)) {
    return reject(new Error(`Route has no legs data: ${JSON.stringify(route)}`));
  }
```

Reply:
I splitted it.